### PR TITLE
libopenraw-legacy: pre-Rust version for old systems

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

New port.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

macOS 10A190
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
Making a draft, I want to try building a newer version without Rust, see: https://gitlab.freedesktop.org/libopenraw/libopenraw/-/issues/13
We still gonna need a `-legacy` subport, but likely at a newer version.

**Comment by barracuda156** [comment]
So yeah, the current version builds without Rust with a minimal patch. Will update the PR soon.

**Comment by barracuda156** [comment]
@mascguy Please take a look when you have time.

P. S. While at the moment it could be done as a variant of `libopenraw`, upstream seems to be inclined to make the next major version much more dependent on Rust; see: https://gitlab.freedesktop.org/libopenraw/libopenraw/-/issues/13
Therefore I think it makes better sense to make a legacy port straightaway. That also will make it easier for me to work on it, without affecting the primary, Rust-based one.

File: graphics/libopenraw-legacy/Portfile [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                    port:autoconf-archive \
+                    port:cctools \
+                    port:grep \
+                    port:gsed \
+                    port:gawk
+
+depends_lib         port:libxml2 \
+                    port:curl \
+                    path:include/turbojpeg.h:libjpeg-turbo
+
```
Comment (reneeotten) [inline graphics/libopenraw-legacy/Portfile:51]: the boost PG adds dependencies so you should probably use "-append" when declaring other dependencies.

Comment (barracuda156) [reply graphics/libopenraw-legacy/Portfile:51]: Thank you, will do it. While it seems to work fine without -append, adding it is a safe bet.

Comment (barracuda156) [reply graphics/libopenraw-legacy/Portfile:52]: UPD. Done and rebased to master.

**Comment by barracuda156** [comment]
@reneeotten Anything else to fix here or good to merge?

**Review by mascguy — APPROVED** [review]
Fine with me, sorry for the delay!

**Pull request merged**